common/workunit/workunit.ipp

```diff
@@ -66,7 +65,7 @@ public:
 
 //==========================================================================================
 
-template <typename T, typename IT> struct CachedTags
+template <typename T, typename IT, class BASE> struct CachedTags : public BASE
```

Comment:
BASE isn't really a base class.   You could make the add() functions static and then avoid adding it as a base class.  Probably even cleaner is to use a function argument:

```
template <typename T, typename IT, void (* ADD)(const IPropertyTree *p, IArrayOf<IT> &dst)> struct CachedTags

void addStatistic(const IPropertyTree * p, IArrayOf<IConstWUStatistic> &dst)
{
    dst.append(*new CLocalWUStatistic(&OLINK(*p)));
}

typedef CachedTags<CLocalWUStatistic, IConstWUStatistic, addStatistic> CachedStatistics;

```

Reply:
that is better, didn't think of a template function arg. for some reason.